server/athenian/api/preloading/__main__.py

```diff
@@ -0,0 +1,110 @@
+    parser.add_argument("--persistentdata-db",
+                        default="postgresql://postgres:postgres@0.0.0.0:5432/persistentdata",
+                        help="Pushed and pulled source data that Athenian owns. DB connection "
+                             "string in SQLAlchemy format. This DB is read/write.")
+    parser.add_argument("--detailed", required=False, action="store_true",
+                        help="Whether to also show the memory usage of each column.")
+
+    return parser.parse_args()
+
+
+def main():
+    """Print memory usage for preloaded DataFrames."""
+    args = parse_args()
+    log = logging.getLogger(metadata.__package__)
+    if not check_schema_versions(args.metadata_db,
+                                 args.state_db,
+                                 args.precomputed_db,
+                                 args.persistentdata_db,
+                                 log):
+        return None
+
+    db_conns = {
+        "sdb": args.state_db,
+        "mdb": args.metadata_db,
+        "pdb": args.precomputed_db,
+        "rdb": args.persistentdata_db,
+    }
+    db_options = compose_db_options(args.metadata_db,
+                                    args.state_db,
+                                    args.precomputed_db,
+                                    args.persistentdata_db)
+
+    asyncio.run(_main(db_conns, db_options, args.detailed))
+
+
+async def _main(db_conns: dict, db_options: dict, detailed: bool):
+    tasks, dbs = [], {}
+    for db_shortcut, db_conn in db_conns.items():
+        db = ParallelDatabase(db_conn, **(db_options[f"{db_shortcut}_options"]))
+        tasks.append(db.connect())
+        dbs[db_shortcut] = db
+
+    await gather(*tasks)
+
+    mc = MemoryCachePreloader()
+    mc.preload(detailed, **dbs)
+    await mc._task
+
+    for db_name, db in dbs.items():
+        if not (cache := getattr(db, "cache", None)):
+            continue
+
+        db_total_mem = cache.memory_usage(True, True)
+        _print_summary_line(f"DB: {db_name}", db_total_mem, 1)
+        for name, df in cache.dfs.items():
+            total_mem = df.memory_usage(True, True)
+            _print_summary_line(f"DataFrame: {name}", total_mem, 2)
+            if not detailed:
+                continue
+
+            info = df._mem
+            raw, processed, percentage = info["raw"], info["processed"], info["percentage"]
+            _print_summary_line("Raw", raw["total"], 3)
+            print("--- Raw series")
+            print(raw["series"])
+
+            _print_summary_line("Processed", processed["total"], 3)
+            print("--- Processed series")
+            print(processed["series"])
+
+            _print_summary_line("Percentage", percentage["total"], 3)
+            print("--- Percentage series")
+            print(percentage["series"])
+
+
+def _print_summary_line(title, total, indent_level):
+    prefix = "-" * indent_level
+    print(f"{(prefix + ' ' + title):<35} | [total: {total}]")
```

Comment:
We can use https://docs.python.org/3/library/textwrap.html#textwrap.indent combined with `textwrap.shorten` here.

Reply:
I was not able to use `textwrap` for the summary line, but I refactored a bit the generation of the detailed part.

Reply:
Current output (local data):
```
# DB: mdb                                          | [total: 1.1 MiB]

## DataFrame: prs                                  | [total: 1.1 MiB]

### Raw                                            | [total: 2978425]

### Raw series

    Index                      128
    acc_id                   33976
    node_id                 377975
    number                   33976
    closed                    4247
    closed_at                33976
    created_at               33976
    merged                    4247
    merge_commit_id         557871
    merge_commit_sha        396483
    merged_at                33976
    merged_by_login         273463
    repository_full_name    341840
    updated_at               33976
    user_login              285461
    hidden                    4247
    additions                33976
    deletions                33976
    htmlurl                 460655
    dtype: int64

### Processed                                      | [total: 1152176]

### Processed series

    Index                      128
    acc_id                    4343
    node_id                 135904
    number                   33976
    closed                    4247
    closed_at                33976
    created_at               33976
    merged                    4247
    merge_commit_id         339760
    merge_commit_sha        169880
    merged_at                33976
    merged_by_login           5726
    repository_full_name      6091
    updated_at               33976
    user_login                6186
    hidden                    4247
    additions                33976
    deletions                33976
    htmlurl                 233585
    dtype: int64

### Percentage                                     | [total: 38.684069600543914]

### Percentage series

    Index                   100.000000
    acc_id                   12.782552
    node_id                  35.955817
    number                  100.000000
    closed                  100.000000
    closed_at               100.000000
    created_at              100.000000
    merged                  100.000000
    merge_commit_id          60.902969
    merge_commit_sha         42.846730
    merged_at               100.000000
    merged_by_login           2.093885
    repository_full_name      1.781828
    updated_at              100.000000
    user_login                2.167021
    hidden                  100.000000
    additions               100.000000
    deletions               100.000000
    htmlurl                  50.707145
    dtype: float64
```